logstash-core/lib/logstash/webserver.rb

```diff
@@ -53,18 +53,18 @@ def self.from_settings(logger, agent, settings)
         auth_basic[:password] = required_setting(settings, 'api.auth.basic.password', "api.auth.type")
 
         password_policies = {}
-        password_policies[:mode] = required_setting(settings, 'password_policy.mode', "api.auth.type")
-
+        password_policies[:mode] = required_setting(settings, 'api.auth.basic.password_policy.mode', "api.auth.type")
```

Comment:
```suggestion
        password_policies[:mode] = required_setting_with_changing_default(settings, 'api.auth.basic.password_policy.mode', "api.auth.type", "ERROR")
```

Along with this helper method:

~~~ ruby
def required_setting_with_changing_default(settings, name, trigger, future_value)
  effective_value = required_setting(settings, name, trigger)
  if !settings.set?(name)
    deprecation_logger.deprecated("The default value of `#{name}` will change to `#{future_value}` in a future release of Logstash. If you wish to lock-in the current behavior, please set `#{name}: #{effective_value}`")
  end
  effective_value
end
~~~

Reply:
I will follow up on this. Tried to make `deprecation_logger ` available in `webserver.rb` but it seems it requires additional settings (for me at least investigation how I can make it available). But overall, I agree that after some point we will change the setting and still we get deprecation.